core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -0,0 +1,267 @@
+      else stage.commit(offset)
+  }
+
+  trait Committer {
+    def commit(offset: Consumer.Offset): Future[Done]
+    def commit(offsets: Map[TopicPartition, OffsetAndMetadata]): Future[Done]
+  }
+
+  final case class CommittableOffsetBatchImpl(override val offsets: Map[TopicPartition, OffsetAndMetadata])(stage: Option[Committer])
+      extends CommittableOffsetBatch {
+
+    def updated(committableOffset: Consumer.CommittableOffset): CommittableOffsetBatch = {
+      val s = stage match {
+        case s @ Some(_) => s
+        case None =>
+          committableOffset match {
+            case c: CommittableOffsetImpl => Option(c.stage)
+            case _ => throw new IllegalArgumentException(
+              s"Unknow CommittableOffset, got [${committableOffset.getClass.getName}], " +
+                s"expected [${classOf[CommittableOffsetImpl].getName}]"
+            )
+          }
+      }
+
+      val offset = committableOffset.offset
+      // committed offset should be the next message the application will consume, i.e. +1
+      CommittableOffsetBatchImpl(offsets.updated(
+        new TopicPartition(offset.topic, offset.partition),
+        new OffsetAndMetadata(offset.offset + 1)
+      ))(s)
+    }
+
+    override def commit(): Future[Done] = {
+      if (offsets.isEmpty)
+        Future.successful(Done)
+      else {
+        stage match {
+          case Some(s) => s.commit(offsets)
+          case None =>
+            Future.failed(new IllegalStateException(s"commit not supported, $offsets"))
+        }
+      }
+    }
+  }
+
+}
+
+/**
+ * INTERNAL API
+ */
+private[kafka] class ConsumerStage[K, V](consumerProvider: () => KafkaConsumer[K, V])
+    extends GraphStageWithMaterializedValue[SourceShape[Consumer.CommittableMessage[K, V]], Consumer.Control]
+    with LazyLogging {
+  import ConsumerStage._
+
+  val out = Outlet[Consumer.CommittableMessage[K, V]]("messages")
+  override val shape = SourceShape(out)
+
+  override def createLogicAndMaterializedValue(inheritedAttributes: Attributes) = {
+    val consumer = consumerProvider()
+    var stopping = false
+    val stoppedPromise = Promise[Done]
+
+    val logic = new TimerGraphStageLogic(shape) with Committer with Consumer.Control {
+      private case object Poll
+      // FIXME move out settings
+      private val POLL_INTERVAL = 100.millis
+      private val POLL_DATA_TIMEOUT = 50.millis
+      private val POLL_COMMIT_TIMEOUT = 1.millis
+      private val COMMIT_TIMEOUT = 10.seconds
+      private var awaitingConfirmation = 0L
+      private var buffer: Iterator[Consumer.CommittableMessage[K, V]] = Iterator.empty
+      private var pollScheduled = false
+
+      private def schedulePoll() = {
+        if (!pollScheduled) {
+          pollScheduled = true
+          scheduleOnce(Poll, POLL_INTERVAL)
+        }
+      }
+
+      private def poll() = {
+        def setupConsumer() = {
+          // TODO do we really need to support change of assignments after startup?
```

Comment:
guys, it is here for other reason.

Since kafka consumer is event loop based to get commit confirmation you should call `poll` even if you do not have a demand. So you should mark that you do not interested in new messages, but interested to get commit confirmation